From ebd43ed9a950e8639320caa6dc4419f696455c19 Mon Sep 17 00:00:00 2001
From: Kurt McKee <contactme@kurtmckee.org>
Date: Wed, 20 Aug 2025 09:32:27 -0500
Subject: [PATCH] Fix a test failure caused by an internal docutils `open()`
 call

docutils 0.22 introduced an RST stylesheet feature
that must be able to open files during RST-to-HTML rendering.

The original design of the `test_cli_explicit_format` test
patched `pathlib.Path.open()` so it would always return the
same open file instance...which was always closed on first use.
This caused a failure when docutils tried to open `minimal.css`:

```
ValueError: I/O operation on closed file.
```

This change introduces a temporary input file to read from,
rather than attempting to mock `pathlib.Path.open()`.
---
 tests/test_cli.py | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/tests/test_cli.py b/tests/test_cli.py
index dddf5c3..9515554 100644
--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -44,13 +44,14 @@ def test_cli_invalid_format():
         main(["no-file.invalid"])
 
 
-def test_cli_explicit_format(input_file):
+def test_cli_explicit_format(input_file, tmp_path):
     fmt = input_file.suffix.lstrip(".")
-    with input_file.open() as fp, \
-            mock.patch("pathlib.Path.open", return_value=fp), \
-            mock.patch("builtins.print") as print_:
-        main(["-f", fmt, "no-file.invalid"])
-        print_.assert_called_once()
+
+    temp_input = tmp_path / "invalid.invalid"
+    temp_input.write_text(input_file.read_text())
+
+    with mock.patch("builtins.print") as print_:
+        main(["-f", fmt, str(temp_input)])
         (result,), _ = print_.call_args
 
     with input_file.with_suffix(".html").open() as fp:
